src/charm.py

```diff
@@ -157,6 +173,17 @@ def process_unremoved_units(self, event) -> None:
                 logger.error("Deferring process_unremoved_units: error=%r", e)
                 event.defer()
 
+    def _on_mongodb_relation_joined(self, event: ops.charm.RelationJoinedEvent) -> None:
+        """Add peer to replica set.
+
+        Args:
+            event: The triggering relation joined event.
+        """
+        if self.unit.is_leader():
+            self.update_relation_data()
```

Comment:
it would be good to also do this for `relation-departed-events` (or you can do that in a separate follow up PR, if you do this in a separate PR it would be best to update your integration tests as to only test the adding units scenario :) )